presto-main/src/main/java/com/facebook/presto/operator/Driver.java

```diff
@@ -324,10 +324,20 @@ private void processNewSource(TaskSource source)
                 // this state change by calling isFinished
                 return NOT_BLOCKED;
             }
-            return processInternal();
+            return processInternalAndHandleRevocableMemory();
         }
     }
 
+    private ListenableFuture<?> processInternalAndHandleRevocableMemory()
```

Comment:
I don't like methods with `And` in the name. I would have another method `revokeMemoryIfNecessary` and the just call them both in a sequence